src/lib/sample_metadata.rs

```diff
@@ -306,11 +322,13 @@ pub fn validate_samples(
         }
 
         // If we have more than one sample, or we have one sample with an actual barcode
-        // then add in the undetermined sample.
+        // then add in the undetermined sample.  The ordinal must be larger than the maximum
+        // ordinal of an existing sample.
+        let undetermined_ordinal = samples.iter().map(|s| s.ordinal).max().unwrap() + 1;
```

Comment:
Any reason not to use `previous_ordinal` here instead of taking the max again?

Reply:
Good idea.